kv/dist_sender.go

```diff
@@ -464,16 +464,24 @@ func (ds *DistSender) sendRPC(desc *proto.RangeDescriptor,
 		a.Header().Replica = *replicaMap[addr.String()]
 		return a
 	}
-	firstReply := true
+	// RPC send will use go routine to send request and fill reply, and there is
+	// no lock for reply varible, so it may have race case which just merged
+	// reply got overwritten by RPC returned value if reply itself as RPC parameter
+	// in first try when there are multiple address to try.
+	// So here reply itself is not used as RPC parameter if there is more
+	// than 1 address to try.
```

Comment:
I struggled with this comment. If the below expresses what you mean above, consider using that instead:

> RPCs are sent asynchronously and there is no synchronized access to the reply object, so we use the originally supplied reply as a template for the actual calls. Otherwise there is a race between the call returning (and filling in the reply) and `rpc.Send` cloning it for a new attempt (except when there is only one address to try, which is not a relevant case).